Lib/functools.py

```diff
@@ -30,7 +30,7 @@
 # wrapper functions that can handle naive introspection
 
 WRAPPER_ASSIGNMENTS = ('__module__', '__name__', '__qualname__', '__doc__',
-                       '__annotations__', '__co_annotations__')
+                       '__co_annotations__', '__annotations__')
```

Comment:
This unbroke test_functools, but I feel like the order shouldn't matter.

Reply:
Good call!  I agree.

These test failures were symptoms of a bug in `func_set_co_annotations()`: if you assign to `__co_annotations__` on a function, it clears `__annotations__`, *even if you're setting `__co_annotations__` to `None`.  I assume the implementations on modules and types have the same bug.

I'll go fix that.  Please revise your PR to undo this change.